src/lib/features/ai/ai-service.ts

```diff
@@ -0,0 +1,85 @@
+import type { ChatCompletionMessageParam } from 'openai/resources/chat/completions';
+import type { ChatCompletionRunner } from 'openai/resources/beta/chat/completions';
+import OpenAI from 'openai';
+import type {
+    IAuthRequest,
+    IUnleashConfig,
+    IUnleashServices,
+    Logger,
+} from '../../server-impl';
+import type FeatureToggleService from '../feature-toggle/feature-toggle-service';
+import type { FeatureSearchService } from '../feature-search/feature-search-service';
+import { createFlag } from './tools/create-flag';
+import { toggleFlag } from './tools/toggle-flag';
+import { searchFlag } from './tools/search-flag';
+
+export class AIService {
+    private config: IUnleashConfig;
+
+    private logger: Logger;
+
+    private client: OpenAI | undefined;
+
+    private featureService: FeatureToggleService;
+
+    private featureSearchService: FeatureSearchService;
+
+    constructor(
+        config: IUnleashConfig,
+        {
+            featureToggleService,
+            featureSearchService,
+        }: Pick<
+            IUnleashServices,
+            'featureToggleService' | 'featureSearchService'
+        >,
+    ) {
+        this.config = config;
+        this.logger = config.getLogger('features/ai/ai-service.ts');
+        this.featureService = featureToggleService;
+        this.featureSearchService = featureSearchService;
+    }
+
+    getClient(): OpenAI {
+        if (this.client) {
+            return this.client;
+        }
+
+        const apiKey = this.config.openAIAPIKey;
+        if (!apiKey) {
+            throw new Error(
+                'Unleash AI is unavailable: Missing OpenAI API key',
+            );
+        }
+
+        this.client = new OpenAI({ apiKey });
+        return this.client;
+    }
+
+    chat(
+        messages: ChatCompletionMessageParam[],
```

Comment:
What's the intent here? Does the chat history live on the UI at the moment? Can't quite see how this works behind a load balancer

Reply:
We're still on the early stage of this project where we're mostly just migrating code from the hackathon branch and polishing it a bit. So for now the chat history still lives in the UI. We have another task for later that moves this ownership to the server.

Reply:
In the same vein, I'm pretty sure our `toggle-flag` tool is not respecting change requests currently, which is another task we have for later.